operators/src/pro/ml/xgboost.rs

```diff
@@ -0,0 +1,560 @@
+use std::mem;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use futures::{future, stream, StreamExt, TryStreamExt};
+use geoengine_datatypes::hashmap;
+use geoengine_datatypes::primitives::{
+    ClassificationMeasurement, Measurement, RasterQueryRectangle, SpatialPartition2D,
+};
+use geoengine_datatypes::raster::{
+    BaseTile, Grid2D, GridOrEmpty, GridShape, GridShapeAccess, Pixel, RasterDataType, RasterTile2D,
+};
+use rayon::prelude::ParallelIterator;
+use rayon::slice::ParallelSlice;
+use rayon::ThreadPool;
+use serde::{Deserialize, Serialize};
+use xgboost::{Booster, DMatrix};
+
+use crate::engine::{
+    ExecutionContext, InitializedRasterOperator, MultipleRasterSources, Operator, QueryContext,
+    QueryProcessor, RasterOperator, RasterQueryProcessor, RasterResultDescriptor,
+    TypedRasterQueryProcessor,
+};
+use crate::util::stream_zip::StreamVectorZip;
+use crate::util::Result;
+use futures::stream::BoxStream;
+use RasterDataType::F32 as RasterOut;
+
+use TypedRasterQueryProcessor::F32 as QueryProcessorOut;
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct XgboostParams {
+    model_file_path: String,
+}
+
+pub type XgboostOperator = Operator<XgboostParams, MultipleRasterSources>;
+pub type Tile<P> = BaseTile<GridOrEmpty<GridShape<[usize; 2]>, P>>;
+
+pub struct InitializedXgboostOperator {
+    result_descriptor: RasterResultDescriptor,
+    sources: Vec<Box<dyn InitializedRasterOperator>>,
+    model_file_path: String,
+}
+
+type PixelOut = f32;
+
+#[typetag::serde]
+#[async_trait]
+impl RasterOperator for XgboostOperator {
+    async fn initialize(
+        self: Box<Self>,
+        context: &dyn ExecutionContext,
+    ) -> Result<Box<dyn InitializedRasterOperator>> {
+        let init_rasters = future::try_join_all(
+            self.sources
+                .rasters
+                .iter()
+                .map(|raster| raster.clone().initialize(context)),
+        )
+        .await?;
+
+        let input = init_rasters.get(0).unwrap();
+        let in_desc = input.result_descriptor();
+
+        let out_desc = RasterResultDescriptor {
+            data_type: RasterOut,
+            time: in_desc.time,
+            bbox: in_desc.bbox,
+            resolution: in_desc.resolution,
+            spatial_reference: in_desc.spatial_reference,
+
+            // where do we get this information from?
```

Comment:
Is this the part you are referring to? https://github.com/geo-engine/geoengine/blob/8bf2d34936f0d35506271621e8b9e0d0c23bdb4b/operators/src/processing/expression/mod.rs#L281-L292

Reply:
now that I think about it: inheriting won't make sense I guess. We could take it as a parameter like in the snippet you posted, but then again, I guess the measurement is always the same for a specific learned model, right? So maybe we should store it in (or alongside) the model.json. This would mean users will have to specify the measurement during learning or at least during import of the model. What do you think?

Reply:
I think it would be good to require the class labels, because as of right now I had to use a routine to get the class labels and create a mapping for them which is passed to xgboost in the training process (the corresponding code is not part of the PR). Having a config for that would make things easier.

Reply:
the class labels are always the same for each application of the learner right? So I think they should be part of the model.json and not be supplied on each use of the operator.

Reply:
The `model.json` is generated by XGBoost itself and does not contain information about the class labels. XGBoost always uses integer labels in form of `[0, n_classes)`. I could add the information manually to the model file as part of the routine, but that would still require a definition of the labels at some point?

Reply:
we could
1. hijack the model.json and add some more data, hoping that XGBoost ignores it
2. store a second json next to it with the additional info. Maybe one folder for each learned model with on model.json and metadata.json

Reply:
My personal favourite in this case would be the second option. I'll try to come up with a solution.